gax/src/main/java/com/google/api/gax/core/GaxProperties.java

```diff
@@ -30,18 +30,29 @@
 package com.google.api.gax.core;
 
 import com.google.api.core.InternalApi;
-import java.util.Properties;
 
 /** Provides properties of the GAX library. */
 @InternalApi
 public class GaxProperties {
-  private static final Properties gaxProperties = new Properties();
-  private static final String GAX_PROPERTY_FILE = "/com/google/api/gax/project.properties";
   private static final String DEFAULT_VERSION = "";
+  private static final String GAX_VERSION = getLibraryVersion(GaxProperties.class);
+  private static final String JAVA_VERSION = getLibraryVersion(Runtime.class);
+
+  private GaxProperties() {}
+
+  /** Returns the version of the library that the {@code libraryClass} belongs to */
+  public static String getLibraryVersion(Class<?> libraryClass) {
+    String version = libraryClass.getPackage().getImplementationVersion();
+    return version != null ? version : DEFAULT_VERSION;
+  }
+
+  /** Returns the version of the running JVM */
+  public static String getJavaVersion() {
+    return JAVA_VERSION;
+  }
 
   /** Returns the current version of GAX. */
   public static String getGaxVersion() {
-    String version = PropertiesProvider.loadProperty(gaxProperties, GAX_PROPERTY_FILE, "version");
-    return version != null ? version : DEFAULT_VERSION;
+    return GAX_VERSION;
```

Comment:
I'd like to make sure we try this in api-client-staging before we merge.

Reply:
The point of my comment here is that whatever our versioning logic is, api-client-staging should still compile and pass tests. (In previous changes to versioning, api-client-staging has gotten broken. I just wanted to make sure it wouldn't get broken again.)

Reply:
It should not break. Before it was relying on geenrated properties files. The generation was done as part of build of google-cloud-java. This change makes thing simpler: there is even no properties file generated anymore, it just relies on built-in java featrue for retrieving version of the lib to which the specified class belongs to.